### PR TITLE
Fix #257 : Added following test cases:

### LINKED ISSUE #257: Adding more tests in vagrant
Add Few scenarios in vagrant tests

## Expected Behaviour
Right now vagrant test setup runs only happy workflow. I would like to have following scenarios in vagrant tests:

- [x] One workflow with one worker where all the actions are successful

- [x] One workflow with one worker where an action gets failed

- [x] One workflow with one worker where an action gets timed out

- [x] Two workflows with one worker where all the actions are successful in both the workflows

- [x] Two workflows with one worker where one of the workflow fails and other one runs successfully.

- [x] Two workflows with one worker where one of the workflow timed out and other one runs successfully.

## Possible Solution
Add more test cases on vagrant setup

## Context
This will help us to test the code in different scenarios so that whenever we are trying to cut a release, we should run these test scenarios

### PR DESCRIPTION
**Pull request by parauliya**
1. One workflow with one worker where all the actions are successful
2. One workflow with one worker where an action gets failed
3. One workflow with one worker where an action gets timed out
4. Two workflows with one worker where all the actions are successful in both the workflows
5. Two workflows with one worker where one of the workflow fails and other one runs successfully.
6. Two workflows with one worker where one of the workflow timed out and other one runs successfully.

## Description

This will enhance the end to end testing area.

## Why is this needed

Fixes: #257 

## How Has This Been Tested?
Before adding any test case, I run it locally and then push in this PR.

## How are existing users impacted? What migration steps/scripts do we need?

No impact.

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [x] added unit or e2e tests

### CODE REVIEW

File: test/_vagrant/vagrant_test.go [hunk 1]
```diff
@@ -16,7 +18,7 @@ import (
 	"github.com/tinkerbell/tink/util"
 )
 
-func TestVagrantSetupGuide(t *testing.T) {
+func TestSuccess(t *testing.T) {
```
Comment (gianarb) [inline test/_vagrant/vagrant_test.go:21]: Why?! :) This testcase is written to test for the success of the Vagrant Setup Guide

File: test/_vagrant/vagrant_test.go [hunk 2]
```diff
@@ -130,6 +134,273 @@ func TestVagrantSetupGuide(t *testing.T) {
 	}
 	t.Fatal("Workflow never got to a complite state or it didn't make it on time (5m)")
 }
+func TestTimeout(t *testing.T) {
```
Comment (gianarb) [inline test/_vagrant/vagrant_test.go:137]: `TestTimeout` does not tell me what you are testing against. What about: `TestActionCorrectlyReachesTimeout`

File: test/_vagrant/vagrant_test.go [hunk 3]
```diff
@@ -130,6 +134,273 @@ func TestVagrantSetupGuide(t *testing.T) {
...
+				t.Logf("action %s TIMEDOUT as expected", event.ActionName)
+				return
+			}
+		}
+		time.Sleep(5 * time.Second)
+	}
+	t.Fatal("Workflow never got to a complite state or it didn't make it on time (5m)")
+}
+
+func TestFailed(t *testing.T) {
```
Comment (gianarb) [inline test/_vagrant/vagrant_test.go:271]: This is too generic, tell me what are you testing here, which failure.

**Comment by gianarb** [comment]
Overall it looks good, I think the name of the functions should better explain what are you testing. Timeout?! Failure?! Too generic.

I am also happy to know if you find writing tests comfortable with the vagrant framework

**Comment by parauliya** [comment]
@gianarb I have updated the function names. Please have a look.

**Comment by gianarb** [comment]
@parauliya can you make the timeout to fail after 5s and not 10s please? Go has a strong timeout by default that starts after 10s and I think your test fails before the timeout takes action

**Comment by parauliya** [comment]
> @parauliya can you make the timeout to fail after 5s and not 10s please? Go has a strong timeout by default that starts after 10s and I think your test fails before the timeout takes action

@gianarb my timeout is 6s for the second action. First action complete succesfully and second action gets timeout after 6s.

**Comment by gianarb** [comment]
Can you rebase and squash?! I think with the new timeout your tests should work in CI as well

**Comment by gianarb** [comment]
Currently exploring #277 .

**Comment by Raj-Dharwadkar** [comment]
I see conflicts here so please can you rebase to master?

**Comment by tstromberg** [comment]
@parauliya - Thank you for your work on this. Do you mind if I close the PR until it's ready to be looked at again? It looks like it's doing the right thing.

Once you are able to make the changes, please re-open it and we'll make sure to review the PR in a more timely manner. I'm looking forward to it.

**Pull request closed without merging**